src/objc_func.c

```diff
@@ -0,0 +1,94 @@
+
+/* Compiler implementation of the D programming language
+ * Copyright (c) 2015 by Digital Mars
+ * All Rights Reserved
+ * written by Walter Bright
+ * http://www.digitalmars.com
+ * Distributed under the Boost Software License, Version 1.0.
+ * http://www.boost.org/LICENSE_1_0.txt
+ * https://github.com/D-Programming-Language/dmd/blob/master/src/objc_func.c
+ */
+
+#include "aggregate.h"
+#include "attrib.h"
+#include "declaration.h"
+#include "id.h"
+#include "objc.h"
+#include "scope.h"
+
+// MARK: Ojbc_FuncDeclaration
+
+Ojbc_FuncDeclaration::Ojbc_FuncDeclaration(FuncDeclaration* fdecl)
+{
+    this->fdecl = fdecl;
+    selector = NULL;
+}
+
+// MARK: semantic
+
+void objc_FuncDeclaration_semantic_setSelector(FuncDeclaration *self, Scope *sc)
+{
+    if (!self->userAttribDecl)
+        return;
+
+    Expressions *udas = self->userAttribDecl->getAttributes();
+    arrayExpressionSemantic(udas, sc, true);
+
+    for (size_t i = 0; i < udas->dim; i++)
+    {
+        Expression *uda = (*udas)[i];
+        assert(uda->type);
+
+        if (uda->type->ty != Ttuple)
+            continue;
+
+        Expressions *exps = ((TupleExp *)uda)->exps;
+
+        for (size_t j = 0; j < exps->dim; j++)
+        {
+            Expression *e = (*exps)[j];
+            assert(e->type);
+
+            if (e->type->ty == Tstruct)
+            {
+                StructLiteralExp *literal = (StructLiteralExp *)e;
+                assert(literal->sd);
+
+                if (strcmp(Id::udaSelector->string, literal->sd->toPrettyChars()) == 0)
```

Comment:
Another option is to move this matching code to `StructDeclaration::StructDeclaration`, and store a reference in a static variable as is done for moduleinfo.  (and lots of similar stuff for ClassDeclaration).  This has the benefit of completely avoiding the check when core.attribute isn't imported.

If you did create a `isCoreModule("attribute")` it could be reused in other places, like detecting intrinsics.